src/Codeception/Coverage/Subscriber/LocalServer.php

```diff
@@ -201,6 +201,13 @@ protected function startCoverageCollection($testName)
             // we need to separate coverage cookies by host; we can't separate cookies by port.
             $cookieDomain = isset($c3Url['host']) ? $c3Url['host'] : 'localhost';
         }
+        
+        $cookieParams = [];
+        if ($cookieDomain !== 'localhost') {
+            $cookieParams['domain'] = $cookieDomain;
+        }
+
+        $this->module->setCookie(self::COVERAGE_COOKIE, $value, $cookieParams);
 
         $this->module->setCookie(self::COVERAGE_COOKIE, $value, ['domain' => $cookieDomain]);
```

Comment:
You missed the old setCookie line here, please delete it.

Reply:
Thanks - not sure how I missed that.